src/components/FlowChart/FlowSelection.jsx

```diff
@@ -1,56 +1,118 @@
-import React from "react";
+import React, { useState } from "react";
 import { Dropdown } from "react-bootstrap";
 
+function SelectionIndicator({ selectedId, itemId }) {
+  return selectedId === itemId ? (
+    <i className="material-icons text-primary">radio_button_checked</i>
+  ) : (
+    <i className="material-icons">radio_button_unchecked</i>
+  );
+}
```

Comment:
The component should not be reponsible to decide if it's selected.
Why should it know about some selectedId if it might belong to some other Item?
If you decide this one level up, it's much clearer.